### ISSUE TITLE
Exclude WebSocket Requests from "slow response time" rules of SmartDetection

### ISSUE THREAD

**Issue by ndreisg** [microsoft/ApplicationInsights-dotnet#1379]
Hi there,

my WebApp handles WebSocket requests as well as normal HTTP requests.
These days, I received some "slow response time" notification emails from AI for this WebApp.
The response time of WebSocket Requests is naturally very high (time between open and close handshake).

Is it possible to exclude WebSocket requests from SmartDetection, so that I only get notifications if the response times of the normal HTTP requests are slow?

Best regards
Alex

**Comment by jeffputz** [comment]
This does seem like something that should be configurable out of the box.

**Comment by rickdgray** [comment]
Same issue. This should be accounted for. App Insights treats this as an issue but this behavior is expected.

**Comment by rickdgray** [comment]
Bumping because this should be fixed

**Comment by jeffputz** [comment]
Boing

**Comment by igristov** [comment]
4 years old problem and no solution yet?
Basically, "metricNamespace":"Microsoft.Web/sites","metricName":"HttpResponseTime" should not include WebSocket connections because they can last for hours and it is not what you expect to monitor or trigger alerts on.

**Comment by cijothomas** [comment]
_"metricNamespace":"Microsoft.Web/sites","metricName":"HttpResponseTime"_

^ This is not something produced by the Application Insights SDK.

**Comment by Zurina** [comment]
Any update on this issue?

**Comment by cijothomas** [comment]
https://github.com/microsoft/ApplicationInsights-dotnet/pull/2372/files Fixed SDK to exclude long running SignalR connections.

SmartDetection is not part of this SDK.
"metricNamespace":"Microsoft.Web/sites","metricName":"HttpResponseTime" - this metric is also not part of this sdk, so unfortunately, can't help.

**Comment by ndreisg** [comment]
You could help by forwarding the issue to the right team

**Comment by Zurina** [comment]
Okay, can you point me in the right direction with a link?

**Comment by cijothomas** [comment]
I do not know if teams outside of SDK have Github repo for reporting issues.
If you see issue with application insights sdk produced metrics after https://github.com/microsoft/ApplicationInsights-dotnet/pull/2372/files, this is the right forum.

If it is about smartdetection - my only suggestion is to open Azure support ticket.
"metricNamespace":"Microsoft.Web/sites","metricName":"HttpResponseTime" - If this is about who is producing this metric, my best guess is, its automatically done by Azure WebApps, so you need to create a support ticket for that too.

**Comment by jeffputz** [comment]
It's pretty weird when someone inside the company asks someone outside of the company to influence a product decision, especially going through support, which I suspect we've all had, at best, mediocre interactions with.

**Comment by rafaelgrilli92** [comment]
The issue is still happening and took me a while to understand that it wasn't a real problem with our application and I can't believe it hasn't been looked at yet after all these years...

**Comment by cijothomas** [comment]
> The issue is still happening and took me a while to understand that it wasn't a real problem with our application and I can't believe it hasn't been looked at yet after all these years...

https://github.com/microsoft/ApplicationInsights-dotnet/pull/2372/files Has fixed this.

**Comment by rafaelgrilli92** [comment]
> > The issue is still happening and took me a while to understand that it wasn't a real problem with our application and I can't believe it hasn't been looked at yet after all these years...
> 
> https://github.com/microsoft/ApplicationInsights-dotnet/pull/2372/files Has fixed this.

Was this fix released? I have the latest SDK for .Net (2.4.3) and I'm still getting very high response times on application insights while the response time of the rest of the application seems to be ok. Any suggestions?
![image](https://github.com/microsoft/ApplicationInsights-dotnet/assets/21044452/a29791bc-fbe8-4225-9c5b-48a300787c3a)

**Comment by akeijzer11** [comment]
is this issue still open? @rafaelgrilli92 can you tell us if the issue is fixed in the updated AppInsights package?

we are at the latest SDK for .Net (8.0.6) and we too are seeing these response times, our overall average response times are not usable anymore.
we use Azure.Monitor.OpenTelemetry.AspNetCore (1.2.0)
![image](https://github.com/microsoft/ApplicationInsights-dotnet/assets/25773759/0c38e45a-d8fe-4fd6-8610-6c346e55ba3e)


can we assume that OpenTelemetry doesn't support this fix even though adopting OpenTelemetry is ["recomended"](https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-enable?tabs=aspnetcore#should-i-use-opentelemetry-or-the-application-insights-sdk)?

the 'http.long_running' property is included in the request:
![image](https://github.com/microsoft/ApplicationInsights-dotnet/assets/25773759/edb62303-ecb9-44f1-a3c6-df1c5afe3304)

**Comment by rafaelgrilli92** [comment]
@akeijzer11 not really. I ended up setting up the Azure SignalR instead of using our backend server to handle the connections, so now the weird response times are gone.

**Comment by cijothomas** [comment]
> we use Azure.Monitor.OpenTelemetry.AspNetCore (1.2.0)

I don't think OTel has fixed this, would be good to report this in the AzureMonitor OpenTelemetry repo https://github.com/Azure/azure-sdk-for-net/issues/new/choose

@rajkumar-rangaraj FYI, who owns the AzureMonitor Distro share plans for addressing this in OTel distros.

This is fixed in Application Insights.

**Comment by zhimin-z** [comment]
This issue is stale because it has been open 300 days with no activity. Remove stale label or this will be closed in 7 days. Commenting will instruct the bot to automatically remove the label.

**Comment by SDBennyM** [comment]
This issue still exists. Proxying over my backend towards Azure PubSub.

**Comment by mattsains-msft** [comment]
Hello everyone. This issue makes sense to me, and I've noticed there's a lot of activity in a very similar issue that is about long running operations in general, here: https://github.com/microsoft/ApplicationInsights-dotnet/issues/2746

Once that functionality is available, marking WebSocket requests in this way would make a lot of sense. To try to keep the conversation in one place, I am going to close this issue as a duplicate of that one. Let's move the conversation there.

One thing you'll find in that related issue is that we're waiting on an OpenTelemetry Semantic Convention proposal to be accepted and implemented: https://github.com/open-telemetry/semantic-conventions/issues/1503 Contributing your voice and how that convention would be useful to use with WebSockets monitoring will help the OpenTelemetry folks to understand the use case and necessity, so I encourage you to engage/subscribe to that thread.
